CONTRIBUTING.md

```diff
@@ -4,4 +4,415 @@ Thanks for considering contributing to Composer!
+1.  Each function that needs a docstring must have its input arguments and return statement (if not None) annotated.
+1.  Each argument annotation should include the type. If the argument has a default value, the type annotation should
+    specify "optional", and the docstring should say the default value. There should only be one line for the "Returns:"
+    section, even if a tuple is returned. For example:
+
+    ```python
+
+    def foo(bar: int):
+        """Foo.
+
+        Args:
+            bar (int): Required bar.
+        """
+        ...
+
+    def foo2(bar: int = 42):
+        """Foo2.
+
+        Args:
+            bar (int, optional): Optional bar. (default: ``42``)
+        """
+        ...
+    
+    def foo3(bar: Optional[int] = None):
+        """Foo3.
+
+        Args:
+            bar (int, optional): Optional bar. (default: ``None``)
+        """
+        ...
+
+    def foo4(bar: Union[int, str] = 42):
+        """Foo4.
+
+        Args:
+            bar (int | str, optional): Optional bar. (default: ``42``)
+        """
+        ...
+
+    def foo5(bar: int) -> int:
+        """Foo5.
+
+        Args:
+            bar (int): Required bar.
+
+        Returns:
+            int: Description of return statement.
+        """
+        ...
+
+    def foo6(bar: int) -> Tuple[int, str]:
+        """Foo6.
+
+        Args:
+            bar (int): Required bar.
+
+        Returns:
+            Tuple[int, str]: Description of return statement.
+        """
+        ...
+    ```
+
+1.  For examples in docstrings, use `.. doctest::` or
+    `.. testcode::` . See the [Sphinx Doctest Extension](https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html)
+    for all of the available directives. Do not use `.. code-block::` for Python examples, as they are untested.
+
+    Any test fixtures for doctests should go in [docs/source/doctest_fixtures.py](docs/source//doctest_fixtures.py)
+    or in a `.. testsetup::` block.
+
+    For example:
+    ```python
+    def my_function(x: Optional[torch.Tensor]) -> torch.Tensor:
+        """blah function
+
+        Args:
+            input (torch.Tensor, optional): Your guess.
+
+        Returns:
+            torch.Tensor: How good your input is.
+
+        Raises:
+            ValueError: If your input is negative.
+
+        Example:
+            .. testsetup::
+
+                # optional setup section, not shown in docs
+                import torch
+                x = torch.randn(42)
+                
+
+            .. testcode::
+
+                # shown in docs; runs after testsetup
+                my_function(x)
+        """
+        ...
+    ```
+
+    To check doctests, run:
+    
+    ```bash
+    cd docs && make clean && make html && make doctest
+    ```
+
+## 8. Hparams and Config Classes
+
+Each class that could be initialized by the hparams will need its configuration variables defined in
+[yahp](https://github.com/mosaicml/yahp) dataclass.
```

Comment:
Not sure what this sentence means. What does it mean to be "initialized by the hparams"? How do I know if a class I write could be initialized in this way? also missing "a" before "yahp".

Also, will future code have an associated yahp dataclass? If not, can maybe just rm this section